File: people/app/src/test/java/ru/ifmo/se/dating/people/api/HttpMonitoringApiTest.java
package ru.ifmo.se.dating.people.api;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.web.client.TestRestTemplate;
import ru.ifmo.se.dating.people.PeopleTestSuite;

public final class HttpMonitoringApiTest extends PeopleTestSuite {
    @Autowired
    private TestRestTemplate rest;

    @Test
    public void healthcheck() {
        assertEquals(getHealthcheck(), "public");
    }

    private String getHealthcheck() {
        final var url = "http://localhost:8080/api/v1/monitoring/healthcheck";
        final var response = rest.getForEntity(url, String.class);
        return response.getBody();
    }
}

File: people/app/src/test/kotlin/ru/ifmo/se/dating/people/api/HttpMonitoringApiTest.kt
package ru.ifmo.se.dating.people.api

import org.junit.Assert
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.web.client.TestRestTemplate
import ru.ifmo.se.dating.people.PeopleTestSuite

class HttpMonitoringApiTest : PeopleTestSuite() {
    @Autowired
    private lateinit var rest: TestRestTemplate

    @Test
    fun healthcheck() {
        Assert.assertEquals(getHealthcheck(), "public")
    }

    private fun getHealthcheck(): String {
        val url = "http://localhost:8080/api/monitoring/healthcheck"
        val response = rest.getForEntity(url, String::class.java)
        return response.body!!
    }
}
